### DISCUSSION TITLE
Duplication among linters

### DISCUSSION THREAD

**Discussion by sahidvelji** [golangci/golangci-lint#4661] [Q&A] [answered]
Does this project attempt to list or keep track of duplication among linters? For example, the `nakedret` linter vs Revive's `bare-return` rule?

https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#bare-return

**Comment by bombsimon** [comment, marked as answer]
Not really. We're aware of most linters overlapping but mostly it's up to the user to pick whatever combination they want. There's been some concerns where some linters have contradicting rules like this one: https://github.com/golangci/golangci-lint/issues/1510. This is mostly a problem when you try to apply fixes.

We generally don't accept full overlaps and for very niche linters we usually suggest to add the differences as configurable features of existing linters.

Since we don't recommend any setup (and probably won't in the future either) we don't track what linters and linter rules overlap with others and leave it up to the users to find out.
